Lib/test/test_compile.py

```diff
@@ -846,6 +846,15 @@ def continue_in_while():
             self.assertEqual(None, opcodes[1].argval)
             self.assertEqual('RETURN_VALUE', opcodes[2].opname)
 
+    def test_unloop_break_continue(self):
+        for stmt in ('break', 'continue'):
+            with self.subTest(stmt=stmt):
+                with self.assertRaises(SyntaxError) as err_ctx:
+                    source = f"with object() as obj:\n    {stmt}"
+                    compile(source, f"<unloop_{stmt}>", "exec")
+                exc = err_ctx.exception
+                self.assertEqual(exc.lineno, 2)
```

Comment:
There are some tests for break and continue outside loops in Lib/test/test_syntax.py, I think the lineno check could be added there. For instance, test_break_outside_loop calls self._check_error() but doesn't pass an expected lineno, though it could.  And a similar test for continue could be added next to it.

Reply:
Done!